qoqo/src/noise_models/continuous_decoherence.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright © 2023 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use pyo3::prelude::*;
+use qoqo_macros::noise_model_wrapper;
+use roqoqo::noise_models::{ContinuousDecoherenceModel, NoiseModel};
+use struqture;
+use struqture_py::spins::PlusMinusLindbladNoiseOperatorWrapper;
+
+/// Noise model representing a continuous decoherence process on qubits.
+///
+/// This noise model assumes that all qubits are constantly experiencing
+/// decoherence over time (e.g. due to coupling to the environment).
+/// The noise for each qubit can be different but only single qubit noise is
+/// included in the model.
+///
+/// Noise is given by the rates of the Lindblad equation.
+/// The Lindblad equation is a so-called master equation for the time evolution of a density matrix.
+/// For the example of a single qubit the non-coherent part of the Lindblad equation can take the following form:
+///
+/// d/dt * ρ = Σ Mij * Li * ρ * Lj† - 1/2 * ( Lj† * Li * ρ + ρ * Lj† * Li),
+///
+/// where the indices i and j run from 0 to 2
+///
+/// with L0 = σ+, L1 = σ- and L3 = σz.
+///
+/// Here the genreal incoherent part of the Lindblad equation is internally represented by a [struqture::spins::PlusMinusLindbladNoiseOperator].
+///
+/// To create a complex decoherence model first create the Lindblad noise and then turn it into a ContinuousDecoherenceModel.
+/// For a simple decoherence model, use new to create an empty model and use the add_damping, add_excitation and add_dephasing methods.
+/// For more fine control access the internal lindblad_noise directly and modify it.
+///
+/// Args:
+///
+///     noise_operator (struqture_py.spins.PlusMinusLindbladNoiseOperator): Optional initialisation of Noise Model with given Lindblad operator.
```

Comment:
Should the type be wrapped in an `Optional`?

Reply:
I omitted it here, because the definition is already so long that it stretches readability.